### ISSUE TITLE
brew install osgeo/osgeo4mac/qgis3 fails on macOSX HighSierra

### ISSUE THREAD

**Issue by jradinger** [OSGeo/homebrew-osgeo4mac#520]
When running brew install osgeo/osgeo4mac/qgis3 I run into an error during the make process...
I am working on MacOS High Sierra (10.13.6)

```
==> /usr/local/Cellar/qgis3/3.4.1/libexec/vendor/bin/pip install -v --no-deps --
==> cmake -G Ninja -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-D
==> cmake --build . --target all -- -j 4
Last 15 lines from /Users/Johannes_Radinger/Library/Logs/Homebrew/qgis3/30.cmake:
      QgsPythonUtilsImpl::PyObjectToQString(_object*) in qgspythonutilsimpl.cpp.o
  "_Py_Finalize", referenced from:
      QgsPythonUtilsImpl::exitPython() in qgspythonutilsimpl.cpp.o
  "_Py_Initialize", referenced from:
      QgsPythonUtilsImpl::init() in qgspythonutilsimpl.cpp.o
  "__Py_NoneStruct", referenced from:
      QgsPythonUtilsImpl::getTraceback() in qgspythonutilsimpl.cpp.o
      QgsPythonUtilsImpl::PyObjectToQString(_object*) in qgspythonutilsimpl.cpp.o
      QgsPythonUtilsImpl::getError(QString&, QString&) in qgspythonutilsimpl.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[4865/5104] cd /tmp/qgis3-20181109-98868-1gc7lru/QGIS-final-3_4_1/build/i18n && /usr/local/opt/qt/bin/lrelease -silent /tmp/qgis3-20181109-98868-1gc7lru/QGIS-final-3_4_1/i18n/qgis_cs.ts -qm /tmp/qgis3-20181109-98868-1gc7lru/QGIS-final-3_4_1/build/output/i18n/qgis_cs.qm
[4866/5104] cd /tmp/qgis3-20181109-98868-1gc7lru/QGIS-final-3_4_1/build/i18n && /usr/local/opt/qt/bin/lrelease -silent /tmp/qgis3-20181109-98868-1gc7lru/QGIS-final-3_4_1/i18n/qgis_da.ts -qm /tmp/qgis3-20181109-98868-1gc7lru/QGIS-final-3_4_1/build/output/i18n/qgis_da.qm
[4867/5104] cd /tmp/qgis3-20181109-98868-1gc7lru/QGIS-final-3_4_1/build/i18n && /usr/local/opt/qt/bin/lrelease -silent /tmp/qgis3-20181109-98868-1gc7lru/QGIS-final-3_4_1/i18n/qgis_es.ts -qm /tmp/qgis3-20181109-98868-1gc7lru/QGIS-final-3_4_1/build/output/i18n/qgis_es.qm
ninja: build stopped: subcommand failed.
```
So how can I proceed to get the new QGIS3 installed on my Mac?

**Comment by jjsobanski** [comment]
Same issue on Mojave 10.14

**Comment by mattwigway** [comment]
Can reproduce as well on 10.14.1.

**Comment by mdmadhu** [comment]
I continue to have the same problem on Mojave (10.14.1, and now also on 10.14.2). 

@fjperini, could you please elaborate what you meant when you said you reinstalled everything, and it installed correctly. Would very much like to try out the steps that seem to have worked for you. Thank you.

**Comment by fjperini** [comment]
@mdmadhu I was referring to eliminate all dependencies, verify my homebrew installation and reinstall `qgis3` with all supports.

You should not have problems, if something goes wrong it is very likely due to your homebrew installation, report with `brew doctor` and `brew --config`.

I hope it helps!

**Comment by mdmadhu** [comment]
+1. I get the same thing now

The only way I have been able to move forward is by doing a `brew install —force-bottle qgis3`

**Comment by fjperini** [comment]
@jjsobanski @mdmadhu  You must rebuild `qt5-webkit` since there is a new version of qt (5.12).

My advice is to do a clean installation, the problems you have are very likely due to your installation HOMEBREW (I suggest you delete it and reinstall it) or that you have installed Python out of that environment.

The installation of `qgis3` should have been generated without problems.

**Comment by jjsobanski** [comment]
@fjperini done a clean install of homebrew and encountered exactly the same issue.

**Comment by jjsobanski** [comment]
@nickrobison 
getting this now

```
patching file Source/WTF/wtf/spi/darwin/XPCSPI.h
patching file Tools/qmake/projects/run_cmake.pro
==> /usr/local/opt/qt/bin/qmake ../WebKit.pro -config release -spec macx-clang
==> make
==> make install
Error: An exception occurred within a child process:
  Errno::ENOENT: No such file or directory @ rb_file_s_rename - (/usr/local/Cellar/qt5-webkit/5.12.0/lib/qml/QtWebKit/libqmlwebkitplugin.so, /usr/local/Cellar/qt5-webkit/5.12.0/lib/qml/QtWebKit/libqmlwebkitplugin.dylib)
```

**Comment by nickrobison** [comment]
@jjsobanski Is it not installing the bottle for you?

The error usually occurs when you don't have the Xcode Command Line tools setup correctly. Which you have to do in Xcode -> Preferences -> Locations -> Command Line Tools

**Comment by fjperini** [comment]
@nickrobison @jjsobanski The Homebrew thing said it to rule out problems, if not that ... it must be a problem with Python.

**Comment by jjsobanski** [comment]
@nickrobison so brew was recognising the CLT
```
macOS: 10.14-x86_64
CLT: 10.1.0.0.1.1539992718
Xcode: 10.1
```
but indeed your tip was right - xcode clts were not setup correctly.

nevertheless feels like back to square one. 
```
Last 15 lines from /Users/Sobanski/Library/Logs/Homebrew/qgis3/31.cmake:
  "_Py_Finalize", referenced from:
      QgsPythonUtilsImpl::exitPython() in qgspythonutilsimpl.cpp.o
  "_Py_Initialize", referenced from:
      QgsPythonUtilsImpl::init() in qgspythonutilsimpl.cpp.o
  "__Py_NoneStruct", referenced from:
      QgsPythonUtilsImpl::getTraceback() in qgspythonutilsimpl.cpp.o
      QgsPythonUtilsImpl::PyObjectToQString(_object*) in qgspythonutilsimpl.cpp.o
      QgsPythonUtilsImpl::getError(QString&, QString&) in qgspythonutilsimpl.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by mdmadhu** [comment]
Thanks all, for your efforts!

@nickrobison with v5.12 installed, the `qt5-webkit` error is now gone. But then, I get an error message similar to @jjsobanski (see below)
```
Last 15 lines from /Users/mdmadhusudan/Library/Logs/Homebrew/qgis3/31.cmake:
      QgsPythonUtilsImpl::getTraceback() in qgspythonutilsimpl.cpp.o
      QgsPythonUtilsImpl::PyObjectToQString(_object*) in qgspythonutilsimpl.cpp.o
      QgsPythonUtilsImpl::getError(QString&, QString&) in qgspythonutilsimpl.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Also @nickrobison , yes, I do use Anaconda.

**Comment by nickrobison** [comment]
@jjsobanski @mdmadhu Do you have any other Python versions installed, other than what the system ships or Homebrew installs?

**Comment by mdmadhu** [comment]
> @jjsobanski @mdmadhu Do you have any other Python versions installed, other than what the system ships or Homebrew installs?

Other than what ships with the system and what Homebrew installs, I have an Anaconda install with a couple of Python environments (3.6 and 3.7).

**Comment by jjsobanski** [comment]
@mdmadhu @nickrobison 

I've removed all my python installs (miniconda, native distribution and brew). Cleared my PATH and install python2 and 3 through brew and then it WORKED!

not really the best solution but at least it clarifies what exactly was the problem.

**Comment by nickrobison** [comment]
@jjsobanski Good to hear that it's working! The issue isn't that you have miniconda installed, it's that a non-homebrew Python install is set first in the PATH. The way Homebrew handles dependencies for a specific formula, is that it creates a virtualenv to install them into. But that means that if during installation another Python installation is first in the path, it will use that one, instead of the vendored python. We can probably make some changes to the formula to better guard against this issue, but we may not be able to fully resolve it.

**Comment by mdmadhu** [comment]
> @jjsobanski Good to hear that it's working! The issue isn't that you have miniconda installed, it's that a non-homebrew Python install is set first in the PATH. The way Homebrew handles dependencies for a specific formula, is that it creates a virtualenv to install them into. But that means that if during installation another Python installation is first in the path, it will use that one, instead of the vendored python. We can probably make some changes to the formula to better guard against this issue, but we may not be able to fully resolve it.

@nickrobison: Unlike @jjsobanski, I did not remove the non-Homebrew Python installs, but ensured that the Homebrew install was first in my path, that the Anaconda install was not in my path and nor were other Python installs. Yet, I keep running into the error below:
```
Last 15 lines from /Users/mdmadhusudan/Library/Logs/Homebrew/qgis3/31.cmake:
      QgsPythonUtilsImpl::getError(QString&, QString&) in qgspythonutilsimpl.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
Would be grateful if someone knows how to address this, but, if not, I do understand that it may not be possible to solve this for every custom setup.

**Comment by mdmadhu** [comment]
> Would be grateful if someone knows how to address this, but, if not, I do understand that it may not be possible to solve this for every custom setup.

@nickrobison Just an update to tell you that once I took care of fixing the path (to temporarily remove Anaconda Python out of it), removed an orphaned Python version in `/Library/Frameworks/Python.framework`, and then unliked and re-linked the Homebrew Python, the build process completed without issue. Thank you all for your patience and help!

**Comment by fjperini** [comment]
@nickrobison It occurs to me that a quick solution for this would be to check if it exists `/Library/Frameworks/Python.framework`, and if it exists recommend that it be previously done:

```
export PATH="/usr/local/opt/python/libexec/bin:$PATH"`
```

```
class PythonOther < Requirement
  fatal true

  satisfy(:build_env => false) { !python_other }

  def python_other
    File.exists?("/Library/Frameworks/Python.framework")
  rescue
    return false
  end

  def message
    s = "It will be necessary that Python 3 of homebrew be first in PATH.\n\n"

    s += "Try after doing: $ export PATH=/usr/local/opt/python/libexec/bin:$PATH\n" if python_other
  end
end
```

**Comment by fjperini** [comment]
@nickrobison I will create a pull with this, at least it will be useful for the user (in this situation) to be notified and find a quick solution before starting to build `qgis3`.

**Comment by jradinger** [comment]
I also just want to confirm that I tested the suggested approach:  `unlink python3 && brew link python3 -force` and then checking the PATH. After that QGIS3 it installed without problems. So it seems, as already mentioned by others, related to the installed versions of Python (without having a clue whats the reason behind that).

**Comment by fjperini** [comment]
@jradinger Well, it seems that it will be the solution for the moment.

Once you do `brew unlink python && brew link python --force` did you need to change the PATH?

**Comment by bhive01** [comment]
I did not need to change anything after the unlink && link step. It
installed easily thereafter.

**Comment by fjperini** [comment]
@jradinger @bhive01 I think that this will be enough for the moment

```
class PythonOther < Requirement
  fatal true

  satisfy(:build_env => false) { !python_org && !python_anaconda && !python_conda}

  def python_org
    File.exists?("/Library/Frameworks/Python.framework")
  rescue
    return false
  end

  def python_anaconda
    File.exists?(File.join(Dir.home, "anaconda"))
  rescue
    return false
  end

  def python_conda
    File.exists?(File.join(Dir.home, "miniconda"))
  rescue
    return false
  end

  def message
    s = "It seems that you have installed another version of Python.\n\n"

    s += "It will be necessary that there is no other Python in PATH and/or that it is not linked to HOMEBREW_PREFIX/bin/python3\n\n"

    s += "You installed Python from with the python.org installer.\n" if python_org

    s += "You installed Python from with the anaconda.com installer.\n" if python_anaconda

    s += "You installed Python from with the conda.io installer.\n\n" if python_conda

    s += "1 - Try after doing:

          $ brew unlink python && brew link --force python\n

          $ $PATH (Check that there is no other version)\n\n"

    s += "2 - You may have installed Anaconda or Miniconda anywhere other than $HOME/anaconda or $HOME/miniconda.

          If you installed in another PATH, proceed in the same way (1).\n\n"


    s += "3 - As a last resort, remove all your Python installations. Clean your PATH, install Python through Homebrew and (1).\n\n"

    s += "It can also be helpful

          $ export PATH=/usr/local/opt/python/libexec/bin:$PATH\n\n"
  end
end
```

For more security if there is an `alias python3`

```
venv = virtualenv_create(libexec/'vendor', "#{HOMEBREW_PREFIX}/bin/python3")
```
```
args << "-DPYTHON_EXECUTABLE='#{`#{HOMEBREW_PREFIX}/bin/python3 -c "import sys; print(sys.executable)"`.chomp}'"
```

The two things that could have happened is that there was another Python in PATH and/or that it was linked to `/usr/local/bin/python3`.

**Comment by fjperini** [comment]
This is better, just a warning in case there is another version of Python

```
printf "\n\n\033[31mIt seems that you have installed another version of Python.\e[0m\n\n"

printf "You installed Python from with the python.org installer.\n\n" if File.exists?("/Library/Frameworks/Python.framework")

printf "You installed Python from with the anaconda.com installer.\n\n" if File.exists?(File.join(Dir.home, "anaconda"))

printf "You installed Python from with the conda.io installer.\n\n" if File.exists?(File.join(Dir.home, "miniconda"))

printf "\e[0mIf the installation failed due to the problem reported in \e[32mhttps://github.com/OSGeo/homebrew-osgeo4mac/issues/520\e[0m\n\n"

printf "1 - Try after doing:\n\n"

printf "    $ brew unlink python && brew link --force python\n\n"

printf "    $PATH (Check that there is no other version)\n\n"

printf "2 - You may have installed Anaconda or Miniconda anywhere other than $HOME/anaconda or $HOME/miniconda, proceed in the same way (1).\n\n"

printf "3 - As a last resort, remove all your Python installations. Clean your PATH, install Python through Homebrew and (1).\n\n"

printf "\033[31mYou can constinue, but remember the above\e[0m -  press 'y' to continue: "
prompt = STDIN.gets.chomp
return unless prompt == 'y'
```

The `class PythonOther < Requirement` forced remove all your Python installations.

Now it's just a message

<img width="1020" alt="python" src="https://user-images.githubusercontent.com/21315242/50168852-1eced980-02cb-11e9-9937-2cd34c65a39d.png">

For more security

```
venv = virtualenv_create(libexec/'vendor', "#{HOMEBREW_PREFIX}/opt/python/bin/python3")

args << "-DPYTHON_EXECUTABLE='#{`#{HOMEBREW_PREFIX}/opt/python/bin/python3 -c "import sys; print(sys.executable)"`.chomp}'"
```

The two things that could have happened is that there was another Python in PATH and/or that it was linked to `/usr/local/bin/python3`.

**Comment by luispuerto** [comment]
I think I have a similar problem: 

```shell 
==> Installing osgeo/osgeo4mac/qgis3 --with-grass --with-saga-gis-lts --with-orfeo6
==> Cloning https://github.com/qgis/QGIS.git
Updating /Users/lpuerto/Library/Caches/Homebrew/qgis3--git
==> Checking out branch release-3_4
Already on 'release-3_4'
Your branch is up to date with 'origin/release-3_4'.
HEAD is now at 5364afe4 [themes] Last round of theme fix (hopefully)
==> Patching
patching file cmake/FindQsci.cmake

In case you have installed another version of Python (Anaconda, Miniconda or if you used the installer provided by python.org).

If the installation failed due to the problem reported in https://github.com/OSGeo/homebrew-osgeo4mac/issues/520

Try after doing:

    $ brew unlink python && brew link --force python

The installation will continue, but remember the above.
30.29.28.27.26.25.24.23.22.21.20.19.18.17.16.15.14.13.12.11.10.09.08.07.06.05.04.03.02.01.00.
==> Cloning https://gitlab.orfeo-toolbox.org/orfeotoolbox/qgis-otb-plugin.git
Updating /Users/lpuerto/Library/Caches/Homebrew/qgis3--otb--git
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at a1762b7 respect naming conventions for qgis providers
==> Downloading https://gist.githubusercontent.com/fjperini/009a5dda278c1a10e0758865d149820c/raw/ff5a239ac525d8e760d6f02
Already downloaded: /Users/lpuerto/Library/Caches/Homebrew/downloads/c76aac94b4eac16a2e775205362eafdc47404aaebad529a64279f5eb2917d7d1--OTBAlgorithmProvider.patch
==> Downloading https://gist.githubusercontent.com/fjperini/53f53daf78ab5e203fd81322ea5c3a2f/raw/783b17df86d8e88c41e26a3
Already downloaded: /Users/lpuerto/Library/Caches/Homebrew/downloads/92632cfec6bff357d3923d6ae56eb0af56ac7f8d85ba52babc61c6389b068a95--OTBUtils.patch
==> patch -p1 -i /private/tmp/qgis3-20190113-59504-1fqu0c/OTBAlgorithmProvider.patch
Last 15 lines from /Users/lpuerto/Library/Logs/Homebrew/qgis3/01.patch:
2019-01-13 16:14:49 +0200

patch
-p1
-i
/private/tmp/qgis3-20190113-59504-1fqu0c/OTBAlgorithmProvider.patch

patching file python/plugins/otb/OTBAlgorithmProvider.py
Hunk #1 FAILED at 53.
Hunk #2 FAILED at 172.
Hunk #3 FAILED at 209.
patch unexpectedly ends in middle of line
Hunk #4 succeeded at 134 with fuzz 2 (offset -106 lines).
3 out of 4 hunks FAILED -- saving rejects to file python/plugins/otb/OTBAlgorithmProvider.py.rej

If reporting this issue please do so to (not Homebrew/brew or Homebrew/core):
osgeo/osgeo4mac

These open issues may also help:
Problem when running brew upgrade qGis3 https://github.com/OSGeo/homebrew-osgeo4mac/issues/581
brew install osgeo/osgeo4mac/qgis3 fails ... usr/local/Cellar/qgis3 folder is missing https://github.com/OSGeo/homebrew-osgeo4mac/issues/553
QGIS3 crashes when I click on QGIS → About QGIS  https://github.com/OSGeo/homebrew-osgeo4mac/issues/582
qgis3 formula needs unlinked grass7 https://github.com/OSGeo/homebrew-osgeo4mac/issues/508
does QGIS3 --with-r have to be with "official" homebrew r?  https://github.com/OSGeo/homebrew-osgeo4mac/issues/583
brew install osgeo/osgeo4mac/qgis3 fails on macOSX HighSierra https://github.com/OSGeo/homebrew-osgeo4mac/issues/520
QGIS3 save fails and corrupts app https://github.com/OSGeo/homebrew-osgeo4mac/issues/455
```
 Perhaps it's related to OBT plugin update?

**Comment by fjperini** [comment]
@luispuerto Yes, it is OTB. I already updated the pacth!

**Comment by luispuerto** [comment]
@fjperini great 👍🏼! I understand that I don’t need to pull since you have in a gist, do I?

**Comment by fjperini** [comment]
Yes, this I already solved.
Now I'm about to see a movie, I come back later :joy:

**Comment by fjperini** [comment]
@luispuerto I'll leave the new parch for you to try. Later I apply the changes.

https://gist.githubusercontent.com/fjperini/009a5dda278c1a10e0758865d149820c/raw/1f854f223466a6309b1a9bd1b002595300ef43f6/OTBAlgorithmProvider.patch

https://gist.githubusercontent.com/fjperini/53f53daf78ab5e203fd81322ea5c3a2f/raw/a337ba3467ad5b3ca7e1c6ed3d54c81a06fbe567/OTBUtils.patch

Hug!

**Comment by luispuerto** [comment]
Now it works!! 🎉  perfect! 👍

**Comment by luispuerto** [comment]
I talked too soon: 

![screen shot 2019-01-13 at 22 05 58](https://user-images.githubusercontent.com/22882880/51090099-76078300-177f-11e9-8a80-6446a3b371ec.png)


```
Couldn't load plugin 'otb' due to an error when calling its classFactory() method 

ModuleNotFoundError: No module named 'otb.OtbAlgorithmProvider' 
Traceback (most recent call last):
  File "/usr/local/Cellar/qgis3/3.4.3/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 335, in startPlugin
    plugins[packageName] = package.classFactory(iface)
  File "/usr/local/Cellar/qgis3/3.4.3/QGIS.app/Contents/MacOS/../Resources/python/plugins/otb/__init__.py", line 37, in classFactory
    from .OTBPlugin import OTBPlugin
  File "/usr/local/Cellar/qgis3/3.4.3/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 672, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "/usr/local/Cellar/qgis3/3.4.3/QGIS.app/Contents/MacOS/../Resources/python/plugins/otb/OTBPlugin.py", line 26, in 
    from .OtbAlgorithmProvider import OtbAlgorithmProvider
  File "/usr/local/Cellar/qgis3/3.4.3/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 672, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
ModuleNotFoundError: No module named 'otb.OtbAlgorithmProvider'


Python version: 3.7.2 (default, Jan 12 2019, 20:39:48) [Clang 10.0.0 (clang-1000.11.45.5)] 
QGIS version: 3.4.3-Madeira Madeira, 5364afe4 

Python Path:
/Users/lpuerto/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/gml_application_schema_toolbox/extlibs
/usr/local/Cellar/qgis3/3.4.3/QGIS.app/Contents/MacOS/../Resources/python
/Users/lpuerto/Library/Application Support/QGIS/QGIS3/profiles/default/python
/Users/lpuerto/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins
/usr/local/Cellar/qgis3/3.4.3/QGIS.app/Contents/MacOS/../Resources/python/plugins
/usr/local/opt/gdal2-python/lib/python3.7/site-packages
/usr/local/opt/qgis3/lib/python3.7/site-packages
/usr/local/opt/qgis3/libexec/vendor/lib/python3.7/site-packages
/usr/local/Cellar/qgis3/3.4.3/libexec/vendor/lib/python3.7/site-packages
/Users/lpuerto
/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python37.zip
/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7
/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/lib-dynload
/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages
/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/geos
/usr/local/Cellar/matplotlib/2.2.3/libexec/lib/python3.7/site-packages
/usr/local/Cellar/numpy/1.15.4_2/libexec/nose/lib/python3.7/site-packages
/Users/lpuerto/Library/Application Support/QGIS/QGIS3/profiles/default/python
```

It loads... I've tried to uncheck the plugin in the interface but it crashes.

**Comment by luispuerto** [comment]
I think I'm going to open a new issue. What do you think?

**Comment by fjperini** [comment]
@luispuerto Okay!
I did not have this problem, but I will reinstall everything again to verify it.

You installed `orfeo6` from #584?

**Comment by luispuerto** [comment]
@fjperini No, I didn't download that pull request... I'm building in this moment without orfeo... I can try again later or tomorrow.

**Comment by fjperini** [comment]
Ok!
I will apply the changes in `qgis3`.

**Comment by luispuerto** [comment]
great... 

If you put a couple of pull request I can download them all merge and test them here. 😃

**Comment by fjperini** [comment]
There was a small error in the patch may be the cause. Give me a few minutes and sent the pull!

**Comment by luispuerto** [comment]
OKIDOKI 👍

**Comment by fjperini** [comment]
@luispuerto I pass through here the [formula](https://gist.githubusercontent.com/fjperini/160aeca0b83c0f797c59773e0a81ca08/raw/dd50f7d3a0ef3afc6745b0b4c4387cc300a421b4/qgis3.rb), while verifying that it builds well.

Do not forget to use #584.

**Comment by luispuerto** [comment]
`--with-saga`? You lost me there...

**Comment by fjperini** [comment]
change to  `sip_dir = "HOMEBREW_PREFIX/share/sip"`

```
--- a/cmake/FindPyQt5.py
+++ b/cmake/FindPyQt5.py
@@ -39,7 +39,7 @@
     import os.path
     import sys
     cfg = sipconfig.Configuration()
-    sip_dir = cfg.default_sip_dir
+    sip_dir = HOMEBREW_PREFIX/share/sip
     if sys.platform.startswith('freebsd'):
         py_version = str(sys.version_info.major) + str(sys.version_info.minor)
         sip_dir = sip_dir.replace(py_version, '')
```


or remove that patch, I'm just seeing if that resolves the `couldn't load SIP module -- PyQt5 problem?`, it seems that some have that problem.

p/d: before: `--with-saga-gis-lts`.
